stylix/palette.nix

```diff
@@ -81,11 +112,15 @@ in
         # and not anything indirect such as filling a template, otherwise
         # the output of the palette generator will not be protected from
         # garbage collection.
-        default = pkgs.runCommand "palette.json" { } ''
-          ${lib.getExe cfg.paletteGenerator} \
-            "${cfg.polarity}" \
-            ${lib.escapeShellArg "${cfg.image}"} \
-            "$out"
+        default = pkgs.runCommand "raw-palette.json" { } ''
+          ${lib.getExe pkgs.matugen} \
+            --contrast ${toString cfg.themeGeneration.contrast} \
+            --dry-run \
+            --json rgb \
+            --type ${cfg.themeGeneration.scheme} \
+            image \
+            "${cfg.image}" \
```

Comment:
These variable interpolations can run into issues with string escape syntax.

It is better to (e.g.) use `env` to pass them into the dev builder as environment variables instead of directly interpolating them into the build script.

Internally, this will use `lib.strings.escapeShellVars` or `escapeShellArg` or similar.